### PR TITLE
Add cluster type field in hdinsight filter

### PR DESCRIPTION
**Pull request by duoxu**
Read cluster type in hdinsight cluster manifest and append to every log/metrics sent to OMS.

### CODE REVIEW

**Comment by duoxu** [comment]
@Microsoft/omsagent-devs
@agup006 

Please take a look.

**Comment by duoxu** [comment]
Sorry, I will fix the test.

File: test/code/plugins/hdinsightfilter_plugintest.rb [hunk 1]
```diff
@@ -20,10 +20,11 @@ def create_driver(conf=CONFIG)
...
   def test_filter
     d = create_driver
-    d.instance.command = "echo 'fake_cluster_name'"
+    d.instance.command = "echo '{\"cluster_name\":\"fake_cluster_name\", \"cluster_type\":\"fake_cluster_type\"}'"
     d.run
     record = Hash.new
     emit = d.instance.filter("", "", record)
-    assert_equal("fake_cluster_name\n", emit["ClusterName"])
+    assert_equal("fake_cluster_name", emit["ClusterName"])
+	assert_equal("fake_cluster_type", emit["ClusterType"])
```
Comment (lagalbra) [inline test/code/plugins/hdinsightfilter_plugintest.rb]: nit: use consistent indentation on this line

File: source/code/plugins/filter_hdinsight.rb [hunk 2]
```diff
@@ -11,24 +11,27 @@ class HdinsightFilter < Filter
...
     SCRIPT = BASE_DIR + '/bin/hdinsightmanifestreader.rb '
-	
-	attr_accessor :command
+
+    attr_accessor :command
 
     def configure(conf)
       super
-      @hostname = OMS::Common.get_hostname or "Unknown host"
+      @hostname = OMS::Common.get_hostname or "unknown"
```
Comment (lagalbra) [inline source/code/plugins/filter_hdinsight.rb]: Any reason that this string changed? "unknown" is a little less clear than "Unknown host"

File: test/code/plugins/hdinsightfilter_plugintest.rb (hunk 1 above)
Comment (duoxu) [inline test/code/plugins/hdinsightfilter_plugintest.rb]: Thanks! I will fix this.

File: source/code/plugins/filter_hdinsight.rb (hunk 2 above)
Comment (duoxu) [inline source/code/plugins/filter_hdinsight.rb]: In cluster_type, the default I set to "unknown" if this information cannot be retrieved from our manifest, maybe I should set it to "unknown cluster type", making it consistent with hostname instead.

**Pull request closed without merging**